components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1251,30 +1251,66 @@ where
                 return;
             }
+            Some(ForceLeaderState::WaitTicks { .. }) => {
+                self.fsm.peer.force_leader = None;
+            }
             None => {}
         }
 
-        if self.fsm.peer.is_leader() {
-            warn!(
-                "reject pre force leader due to already being leader";
-                "region_id" => self.fsm.region_id(),
-                "peer_id" => self.fsm.peer_id(),
-            );
-            return;
-        }
-
+        let ticks = if self.fsm.peer.is_leader() {
+            if self.fsm.hibernate_state.group_state() == GroupState::Ordered {
+                warn!(
+                    "reject pre force leader due to already being leader";
+                    "region_id" => self.fsm.region_id(),
+                    "peer_id" => self.fsm.peer_id(),
+                );
+                return;
+            }
+            // wait two rounds of election timeout to trigger check quorum to step down the leader
+            Some(
+                (self.ctx.cfg.raft_election_timeout_ticks * 2)
+                    .saturating_sub(self.fsm.peer.raft_group.raft.election_elapsed)
+                    + 1,
+            )
         // When election timeout is triggered, leader_id is set to INVALID_ID.
         // But learner(not promotable) is a exception here as it wouldn't tick
         // election.
-        if self.fsm.peer.raft_group.raft.promotable()
-            && self.fsm.hibernate_state.group_state() != GroupState::Idle
+        } else if self.fsm.peer.raft_group.raft.promotable()
             && self.fsm.peer.leader_id() != raft::INVALID_ID
         {
-            warn!(
-                "reject pre force leader due to leader lease may not expired";
+            if self.fsm.hibernate_state.group_state() == GroupState::Ordered {
+                warn!(
+                    "reject pre force leader due to leader lease may not expired";
+                    "region_id" => self.fsm.region_id(),
+                    "peer_id" => self.fsm.peer_id(),
+                );
+                return;
+            }
+            // wait one round of election timeout to make sure leader_id is invalid
+            Some(
+                self.ctx
+                    .cfg
+                    .raft_election_timeout_ticks
+                    .saturating_sub(self.fsm.peer.raft_group.raft.election_elapsed)
+                    + 1,
+            )
+        } else {
+            None
+        };
+
+        if let Some(ticks) = ticks {
+            info!(
+                "enter wait ticks";
                 "region_id" => self.fsm.region_id(),
                 "peer_id" => self.fsm.peer_id(),
+                "ticks" => ticks,
             );
+            self.fsm.peer.force_leader = Some(ForceLeaderState::WaitTicks {
+                failed_stores,
+                ticks,
+            });
+            self.reset_raft_tick(GroupState::Ordered);
```

Comment:
Why use `Ordered`? What if it's `Chaos` before?

Reply:
I want to keep ticking now to make sure a possible election timeout or check quorum is triggered within a finite time span.

Reply:
I know why you want to register tick, but I don't understand why change group state.

Reply:
because I want to exit hibernate state if any, otherwise ticking is paused  and`check_force_leader` won't be called, and force leader is blocked for a long time.

Reply:
It still doesn't explain why `Ordered` is used.

Reply:
I don't understand. So which one should it use?

Reply:
It should behave just like accepting a new request.

Reply:
Seems it just sets to `Ordered` when proposing

Reply:
No, when it's follower, it will set it to `Chaos`.